interview-questions.md

```diff
@@ -9,59 +9,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer:So, from what i remember in class i remember we were doing a lot of class/class inheritance for example; 
+                                               class grocery{
+                                              constructor(){
+                                              this.apple = 0.60
+                                                this.orange = 0.70
+                                              let grocery = new grocery()
+                                               console.log(grocery)
+                                        --> grocery { apple: 0.60, orange:0.70}
+  so on my example  i set up a class what is amazing about this if it reference itself a global object but, if you connect this with an object this.apple or this.orange it will refer to the object you are connecting it to.
 
-  Researched answer:
+  Researched answer:The JavaScript this keyword refers to the object it belongs to. ... Alone, this refers to the global object. In a function, this refers to the global object. In a function, in strict mode, this is undefined . In an event, this refers to the element that received the event.
 
 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: From what we did in class we did use yarn but first i would tell you what some information about yarn. Yarn is a package manager you can use it for a lot of things like if you want to make an app, testing, and it is realible, and works fast a lot of developers actually uses yarn cause it is a useful and important tool. For example if you type in yarn jest on your terminal it is helpful for testing, if you type in yarn start it is useful for app applications.
 
   Researched answer:
+Yarn is a new JavaScript package manager that aims to be speedy, deterministic, and secure. See how easy it is to drop yarn in where you were using npm before, and get faster, more reliable installs. [00:01] Yarn is a package manager for JavaScript
 
 
 
 3. What are props in React?
 
-  Your answer:
+  Your answer: i am going to try my best to explain this but from what i learn in class i know props stands for properties and the way the props works it passes data from 1 to another componant. Also, when you use the class method the class is the parent and you are passing the same data to the child.
 
-  Researched answer:
+  Researched answer:React allows us to pass information to a Component using something called props (stands for properties). Props are basically kind of global variable or object
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: oh ok i know this So, from class we did learn jsx and jsx is similar to HTML but the difference jsx is use more for react and syntax and arranges the word in the sentence. I remember it cause i know one of my instructor told we will use jsx a lot in react.
```

Comment:
Yea, React only uses JSX for its HTML lookalike syntax. JSX is a combination of JavaScript and XML (X - HTML).